v2/errorreportingutil/errorreporting_test.go

```diff
@@ -0,0 +1,246 @@
+					assert.Equal(t, tt.expectedMsg, mock.reported[0].Error.Error())
+				} else {
+					assert.Nil(t, mock.reported[0].Error)
+				}
+			} else {
+				assert.Empty(t, mock.reported, "should not report ignored errors")
+			}
+		})
+	}
+}
+
+func TestRecover(t *testing.T) {
+	testErr := errors.New("error panic")
+
+	tests := []struct {
+		name             string
+		panicValue       any
+		expectPanic      bool
+		expectReported   bool
+		validateError    func(t *testing.T, err error)
+		validateStack    func(t *testing.T, stack []byte)
+		validateOutput   func(t *testing.T, output string)
+		expectedPanicMsg string
+	}{
+		{
+			name:           "string panic",
+			panicValue:     "test panic",
+			expectPanic:    true,
+			expectReported: true,
+			validateError: func(t *testing.T, err error) {
+				assert.Contains(t, err.Error(), "test panic")
+			},
+			validateStack: func(t *testing.T, stack []byte) {
+				assert.NotEmpty(t, stack, "should capture stack trace")
+			},
+			validateOutput: func(t *testing.T, output string) {
+				assert.Contains(t, output, "panic: test panic")
+				lines := strings.Split(output, "\n")
+				assert.True(t, len(lines) > 1, "should have multiple lines including stack trace")
+			},
+		},
+		{
+			name:           "error panic",
+			panicValue:     testErr,
+			expectPanic:    true,
+			expectReported: true,
+			validateError: func(t *testing.T, err error) {
+				assert.Equal(t, testErr, err)
+			},
+			validateStack: func(t *testing.T, stack []byte) {
+				assert.NotEmpty(t, stack)
+				stackStr := string(stack)
+				assert.Contains(t, stackStr, "errorreportingutil", "stack trace should contain package name")
+			},
+			validateOutput: func(t *testing.T, output string) {
+				assert.Contains(t, output, "panic: error panic")
+			},
+			expectedPanicMsg: "error panic",
+		},
+		{
+			name:           "no panic",
+			panicValue:     nil,
+			expectPanic:    false,
+			expectReported: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mock := &mockErrorClient{}
+
+			_, err := eutil.Init(
+				context.Background(),
+				eutil.WithProjectID("12345"),
+				eutil.WithClient(mock),
+			)
+			require.NoError(t, err)
+
+			if tt.expectPanic {
+				if tt.expectedPanicMsg != "" {
+					assert.PanicsWithError(t, tt.expectedPanicMsg, func() {
+						defer eutil.Recover()
+						panic(tt.panicValue)
+					})
+				} else {
+					assert.Panics(t, func() {
+						defer eutil.Recover()
+						panic(tt.panicValue)
+					})
+				}
+			} else {
+				assert.NotPanics(t, func() {
+					defer eutil.Recover()
+					// No panic
+				})
+			}
+
+			if tt.expectReported {
+				require.Len(t, mock.reported, 1)
+				if tt.validateError != nil {
+					tt.validateError(t, mock.reported[0].Error)
+				}
+				if tt.validateStack != nil {
+					tt.validateStack(t, mock.reported[0].Stack)
+				}
+			} else {
+				assert.Empty(t, mock.reported, "should not report anything when no panic occurs")
+			}
+		})
```

Comment:
Well this one is really complex :D. There's only 3 cases so how about creating a separated test function for each test with dedicated logic instead of if elsing all into one?